src/cryptography/hazmat/backends/openssl/aead.py

```diff
@@ -256,57 +95,454 @@ def _decrypt(
     tag_length: int,
     ctx: typing.Any = None,
 ) -> bytes:
-    from cryptography.hazmat.primitives.ciphers.aead import AESCCM, AESSIV
+    if _is_evp_aead_supported_cipher(backend, cipher):
+        return _EVPAEAD.decrypt(
+            backend, cipher, nonce, data, associated_data, tag_length, ctx
+        )
+    else:
+        return _EVPCIPHER.decrypt(
+            backend, cipher, nonce, data, associated_data, tag_length, ctx
+        )
 
-    if len(data) < tag_length:
-        raise InvalidTag
 
-    if isinstance(cipher, AESSIV):
-        # RFC 5297 defines the output as IV || C, where the tag we generate is
-        # the "IV" and C is the ciphertext. This is the opposite of our
-        # other AEADs, which are Ciphertext || Tag
-        tag = data[:tag_length]
-        data = data[tag_length:]
-    else:
-        tag = data[-tag_length:]
-        data = data[:-tag_length]
-    if ctx is None:
-        cipher_name = _aead_cipher_name(cipher)
-        ctx = _aead_setup(
-            backend, cipher_name, cipher._key, nonce, tag, tag_length, _DECRYPT
+class _EVPAEAD:
+    """
+    Access to AEADs through EVP_AEAD APIs.
+    """
+
```

Comment:
There's no need for a class here. Please just use a set of free functions.

Reply:
[6ddc6bc](https://github.com/pyca/cryptography/pull/8946/commits/6ddc6bc8abf8e0958252ff3e4eccd87bbdcf4ea0) and [3849acd](https://github.com/pyca/cryptography/pull/8946/commits/3849acde8d7b759b2a6e5b25480f79f53337bc66)